### ISSUE TITLE
InfoNCE function

### ISSUE THREAD

**Issue by FairyFali** [ZhaozwTD/KPE#2]
I'm seeking clarification regarding the implementation of the InfoNCE loss function as shown below:

```
def infonce(loss_func, query, positive_key, negative_keys, temperature=0.01):
    positive_logit = torch.sum(query * positive_key, dim=1, keepdim=True)

    negative_logits = query @ transpose(negative_keys)

    logits = torch.cat([positive_logit, negative_logits], dim=1)
    labels = torch.zeros(len(logits), dtype=torch.long, requires_grad=False).to(query.device)
    loss = loss_func(logits / temperature, labels)

    return loss
```

From my understanding, the label for the positive logit should be 1, with the rest being 0. However, in this code, all elements of labels are initialized to 0. Should the code be modified to set the first element of labels to 1 (i.e., labels[0] = 1) for it to function correctly?

**Comment by FairyFali** [comment]
I see. The labels are all 0, meaning the ground-truth label is the class of 0. Please close this issue.
